### PR TITLE
Shooter

### PR DESCRIPTION
**Pull request by Lewis-Seiden**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package frc.robot.subsystems.shooter;
+import edu.wpi.first.math.geometry.Rotation2d;
+import org.littletonrobotics.junction.AutoLog;
+/** Add your docs here. */
+public interface ShooterIO {
+    @AutoLog
+     public static class ShooterIOInputs {
+     public double drivePositionMeters = 0.0;
+     public double driveVelocityMetersPerSec = 0.0;
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java]: why are there inputs from swerve here? this should just have pivot and flywheel inputs

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     public double PivotAmps = 0;
+     public double PivotMeters = 0.0;
+     public double PivotVoltage = 0.0;
+     public double PivotAmp = 0.0;
+     public double PivotTempc = 0.0;
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java]: remember to use `camelCase`

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+     public double PivotVelocity = 0.0;
+     public Rotation2d turnAbsolutePosition = new Rotation2d();
+     public Rotation2d turnPosition = new Rotation2d();
+     public double turnVelocityRadPerSec = 0.0;
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java]: this is also from swerve

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+     public double FlywheelAmps = 0;
+     public double FlywheelVoltage = 0.0;
+     public double FlywheelAmp = 0.0;
+     public double FlywheelTempc = 0.0;
+     public double FlywheelVelocity = 0.0;
+     public double pivotAppliedVolts = 0.0;
+  
+     }
+     public default void setFlywheelVelocity(final double volts) {} //set voltage
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java]: might want to split this into a `setFlywheelVelocity` that takes rotations per second and a `setFlywheelVoltage` that takes volts

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubystem.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package frc.robot.subsystems.shooter;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.shooter.ShooterIO.ShooterIOInputs;
+
+public class ShooterSubystem extends SubsystemBase {
+ShooterIO PivotIO;
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubystem.java]: make sure to name these ShooterIO

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubystem.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+public ShooterSubystem(ShooterIO PivotIO){
+this.PivotIO =PivotIO;
+PivotInputs = new ShooterIOInputs();
+
+
+
+    }
+@Override        
+public void periodic(){
+PivotInputs = PivotIO.updateInputs();
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubystem.java]: make sure to call `Logger.processInputs`

File: src/main/java/frc/robot/subsystems/shooter/Shooterreal.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+public ShooterIOInputs updateInputs() {
+  ShooterIOInputs inputs = new ShooterIOInputs();
+  inputs.PivotAmps = motor.getSupplyCurrent().getValue();
+  inputs.PivotTempc = motor.getDeviceTemp().getValue();
+  inputs.PivotMeters = motor.getRotorPosition().getValue();
+  inputs.PivotVoltage = motor.getSupplyVoltage().getValue();
+  inputs.PivotVelocity = motor.getVelocity().getValue();
+  inputs.FlywheelAmp = motor.getSupplyCurrent().getValue();
+  inputs.FlywheelTempc = motor.getDeviceTemp().getValue();
+  inputs.FlywheelVoltage = motor.getSupplyVoltage().getValue();
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/Shooterreal.java]: you want to use MotorVoltage not SupplyVoltage

File: src/main/java/frc/robot/subsystems/shooter/Shooterreal.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+public void setFlywheelVelocity(final Rotation2d rotation) {
+  setFlywheelVelocity(
+    pivotAppliedVolts = MathUtil.clamp(volts, -12.0, 12.0));
+    motor.setVoltage(pivotAppliedVolts);
+  }
+ 
+
+private final PIDController pivotController = new PIDController(100.0, 0.0, 0.0);
```
Comment (Lewis-Seiden) [inline src/main/java/frc/robot/subsystems/shooter/Shooterreal.java]: you dont need a PIDController object here, just use a VelocityVoltage control mode

**Review by Awesomeplayer165 — COMMENTED** [review]
looks good so far

**Review by Awesomeplayer165 — APPROVED** [review]

**Pull request merged**